lib/select.c

```diff
@@ -87,15 +87,52 @@ int Curl_wait_ms(int timeout_ms)
 #if defined(MSDOS)
   delay(timeout_ms);
 #elif defined(USE_WINSOCK)
-  Sleep(timeout_ms);
+  /* prevent overflow, timeout_ms is typecast to ULONG/DWORD. */
+#if TIMEDIFF_T_MAX > ULONG_MAX
+  if(timeout_ms > ULONG_MAX)
+    timeout_ms = ULONG_MAX;
```

Comment:
FYI ULONG_MAX is probably equal to INFINITE and I'm having second thoughts on whether that's a good idea. maybe it would be more correct as ULONG_MAX -1 . This is really a corner case I don't see where it could happen.

Reply:
I was not referring to the [INFINITE used for Sleep](https://docs.microsoft.com/en-us/windows/win32/api/synchapi/nf-synchapi-sleep), but the following regarding [select](https://www.man7.org/linux/man-pages/man2/select.2.html) in general:

>               If _timeout_ is specified as NULL, **select()** blocks **indefinitely**
>               waiting for a file descriptor to become ready.

Source: https://www.man7.org/linux/man-pages/man2/select.2.html

Reply:
I think I know understood what you meant and updated the latest commit accordingly.